config/http_config.go

```diff
@@ -88,6 +89,12 @@ func (c *HTTPClientConfig) Validate() error {
 	if c.BasicAuth != nil && (len(c.BearerToken) > 0 || len(c.BearerTokenFile) > 0) {
 		return fmt.Errorf("at most one of basic_auth, bearer_token & bearer_token_file must be configured")
 	}
+	if c.BasicAuth != nil && c.BasicAuth.Username == "" {
+		return fmt.Errorf("basic_auth requires a username")
+	}
+	if c.BasicAuth != nil && (string(c.BasicAuth.Password) != "" && c.BasicAuth.PasswordFile != "") {
+		return fmt.Errorf("don't specify a basic_auth password and password_file")
```

Comment:
minor but it would be good to be consistent with the other error messages, eg `"at most one of basic_auth password & password_file must be configured"`.

Reply:
Good catch. Fixed.